hedera-node/hedera-app/src/main/java/com/hedera/node/app/meta/MonoHandleContext.java

```diff
@@ -57,7 +57,9 @@ public MonoHandleContext(
             @NonNull final OptionValidator optionValidator,
             @NonNull final TransactionContext txnCtx) {
         Objects.requireNonNull(ids);
-        this.nums = () -> ids.newAccountId(PbjConverter.fromPbj(PLACEHOLDER_ID)).getAccountNum();
+        this.nums = () -> ids.newAccountId(PbjConverter.pbjToProto(
```

Comment:
You are replacing a type-safe implementation without reflection with one that is not type safe and uses reflection. Is this really needed?

Reply:
`PbjConverter.fromPbj` throws a runtime exception (`Invalid account ID, no account type!`) because `PLACEHOLDER_ID` doesn't have an account number set, which is acceptable for the old proto objects, but not for PBJ. And, as mentioned below about assigning an account number of -1, it seems dangerous to arbitrarily assign a default account number, even if it's a negative number. I'm not able to think of any at the moment, but are there other options we could try?

Reply:
I don't know either. The whole setup looks weird IMO. We create an arbitrary `AccountID`, which is passed to `EntityIdSource.newAccountId()` to generate another `AccountID` from which we just take the `accountNum`? This does not really sound like the optimal approach. ;)

But we can take a look later. I'd suggest though to do the conversion once separately and use the converted value in the lambda expression.

Reply:
Refactored so that `ids` returns a new account number without all the weirdness